server/src/main/java/org/elasticsearch/health/HealthPeriodicLogger.java

```diff
@@ -458,11 +510,22 @@ private void enable(boolean enabled) {
 
     private void updatePollInterval(TimeValue newInterval) {
         this.pollInterval = newInterval;
-        maybeScheduleJob();
+        // After the health logger is stopped we do not want to reschedule it
+        if (isStoppedOrClosed() == false) {
+            maybeScheduleJob();
+        }
```

Comment:
Do we want to run things if the service is only `INITIALIZED`? Or should we wait until it's fully started? It seems strange to use `isStoppedOrClosed() == false` here, but then `if (isStarted() == false)` for skipping in the `maybeScheduleJob()` definition. Should this just be `if (isStarted()) {...`?

Reply:
The reason I did this was because I was afraid for races. If the service is stopped or closed we are 100% sure that we should not schedule it again. But if the service is initialised, it could potentially be starting in the meantime. We wouldn't want to miss a run then with the correct poll interval. This is why I thought to use stop/closed here to simplify synchronisation policy. Does this make sense?

Reply:
Makes sense, thanks.